src/downloads.jl

```diff
@@ -193,3 +193,123 @@ function ons(timeseries::AbstractString = "L522", dataset::AbstractString = "MM2
     end
     TimeArray(ta, meta = json["description"])
 end
+
+"""
+    struct BoeOpt <: AbstractQueryOpt
+      Datefrom    # the start time
+      Dateto      # the end time
+      UsingCodes  # indicate using series codes
+      CSVF        # "TT" (Tabular with titles), "TN" (Tabular no titles), "CT" (Columnar with- titles) or "CN" (Columnar no titles)
+      VPD         # provisional data is required
+    end
+
+The Bank of England Database API query object.
+
+# Examples
+
+```jl-repl
+julia> t = Dates.today()
+2022-04-06
+
+julia> BoeOpt(Datefrom = t - Year(2), Dateto = t)
+BoeOpt with 5 entries:
+  :Datefrom   => "06/Apr/2020"
+  :Dateto     => "06/Apr/2022"
+  :UsingCodes => "Y"
+  :CSVF       => "TN"
+  :VPD        => "Y"
+```
+"""
+struct BoeOpt <: AbstractQueryOpt
```

Comment:
Usually we use `CamelCase` for modules and type names, not for fields. 
Can we switch to lowercase? Like

```
struct BoeOpt
    date_from::Date
    date_to::Date
    using_codes::Bool
    csvf::String
    vpd::Bool
end
```

Reply:
An AbstractQueryOpt is passed to the HTTP GET command with its fieldnames and values as a query.

I chose fieldnames to be consistent with the API documentation. https://www.bankofengland.co.uk/boeapps/database/Help.asp#CSV

However, I agree that lowercase is standard for fieldnames, and since the parameter names in a HTTP GET query are case-insensitive this shouldn't cause an issue.  However, underscores would not be consistent with AbstractQueryOpt as I would need to change the field names to construct the query.